pkg/pod/pod.go

```diff
@@ -361,6 +365,30 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1beta1.TaskRun, taskSpec
 	return newPod, nil
 }
 
+// applyTaskLevelComputeResources applies the task-level compute resource requirements to the Pod.
+func applyTaskLevelComputeResources(steps []v1beta1.Step, computeResources *corev1.ResourceRequirements, taskSpec v1beta1.TaskSpec) {
+	if taskSpec.StepTemplate != nil {
+		taskSpec.StepTemplate.Resources = corev1.ResourceRequirements{}
+	}
```

Comment:
I don't think you need steptemplate here. Steps are merged with step template and overrides [here](https://github.com/tektoncd/pipeline/blob/3f539724f27a6add8b64d0712e056e4479750975/pkg/pod/pod.go#L136-L149)

Reply:
I see. thanks this context!
let me clean it up.